ui_components/src/home/Home.js

```diff
@@ -0,0 +1,564 @@
+        setDisplayLogin(false);
+        setDisplayRegister(false);
+        setDisplayRules(false);
+        setDisplayTerms(false);
+        setDisplayReset(true);
+    }
+    function handleReset() {
+        document.getElementById('reset_form').submit();
+    }
+    function handleRules() {
+        setDisplayLogin(false);
+        setDisplayRegister(false);
+        setDisplayReset(false);
+        setDisplayTerms(false);
+        setDisplayRules(true);
+    }
+    function handleTerms() {
+        setDisplayLogin(false);
+        setDisplayRegister(false);
+        setDisplayReset(false);
+        setDisplayRules(false);
+        setDisplayTerms(true);
+    }
+    function scrollTo(element) {
+        element.scrollIntoView({ behavior: 'smooth' });
+    } 
+    return (
+        <div className={"home_section login_section"}>
+            <div className="login_center_wrapper">
+                <div className="login_center_inner">
+                    <div className="login_inner_title"><img src="/images/v2/decorations/homepagelogo.png" /></div>
+                    <div className="login_inner_version">0.9 MAKE THINGS LOOK GOOD</div>
+                    <div className="login_inner_input_container">
+                        {displayLogin &&
+                            <form id="login_form" action="" method="post" style={{ zIndex: 1 }}>
+                            <div className="login_input_top">
+                                <div className="login_username_wrapper">
+                                    <label className="login_username_label">username</label>
+                                    <input type="text" name="user_name" className="login_username_input login_text_input" />
+                                </div>
+                                <div className="login_password_wrapper">
+                                    <label className="login_username_label">password</label>
+                                    <input type="password" name="password" className="login_password_input login_text_input" />
+                                </div>
+                                <input type="hidden" name="login" value="login"/>
+                            </div>
+                            {loginMessageText != "" &&
+                                <div className="login_input_bottom">
+                                    <div className="login_message_label">{loginMessageText}</div>
+                                </div>
+                            }
+                            {loginErrorText != "" &&
+                                <div className="login_input_bottom">
+                                    <div className="login_error_label">{loginErrorText}</div>
+                                    <div className="reset_link" onClick={() => handleDisplayReset()}>reset password</div>
+                                </div>
+                            }
+                            </form>
+                        }
```

Comment:
If we go the route of having a `pageToDisplay` var, we could extract each of these to its own component and refactor this to be a switch statement something like

```js
switch(pageToDisplay) {
  case "register":
    return <RegisterForm /* props here */ />;
  case "login":
    return <LoginForm /* props here */ />;
  // ...
}
```

Reply:
Marking as to-do.